src/components/FileList/FileList.js

```diff
@@ -20,17 +31,164 @@ const FileList = ({fileList, studyId}) => {
   const handlePageClick = (e, {activePage}) => {
     setPage(activePage);
   };
-  const sortedFileList = fileList
-    .filter(obj => fileLatestStatus(obj.node) === 'CHN')
-    .concat(fileList.filter(obj => fileLatestStatus(obj.node) !== 'CHN'));
-  const pageCount = Math.ceil(sortedFileList.length / perPage);
-  const pageItems = sortedFileList.slice(
+
+  const [sortMethod, setSortMethod] = useState('');
+  const [sortDirection, setSortDirection] = useState('ascending');
+  const [typeFilterStatus, setTypeFilterStatus] = useState('');
+  const [approvalFilterStatus, setApprovalFilterStatus] = useState('');
+  const statusOptions = Object.keys(versionState).map(state => ({
+    key: state,
+    value: state,
+    text: versionState[state].title,
+    content: <Badge state={state} />,
+  }));
+  const typeOptions = Object.keys(fileTypeDetail).map(type => ({
+    key: type,
+    value: type,
+    text: fileTypeDetail[type].title,
+    content: (
+      <small>
+        <Icon name={`${fileTypeDetail[type].icon || 'question'}`} />
+        {' ' + fileTypeDetail[type].title}
+      </small>
+    ),
+  }));
+  const sortOptions = [
+    {
+      key: 'createDate',
+      value: 'createDate',
+      text: 'Create date',
+    },
+    {
+      key: 'modifyDate',
+      value: 'modifyDate',
+      text: 'Modified date',
+    },
+  ];
+
+  const sortedFileList = () => {
+    var sortedList = [];
+    if (
+      sortMethod === '' &&
+      typeFilterStatus === '' &&
+      approvalFilterStatus === ''
+    ) {
+      sortedList = fileList
+        .filter(obj => fileLatestStatus(obj.node) === 'CHN')
+        .concat(fileList.filter(obj => fileLatestStatus(obj.node) !== 'CHN'));
+    } else {
+      if (sortMethod === 'createDate') {
```

Comment:
It may be cleaner if you move these sort methods to utility functions and map the method to the sort function:
```
const sortsMethods = { 'createDate': sortOldestVersion, 'modifiedDate': sortLatestDate};
const sortedList = fileList.sort(sortMethods[sortMethod])
```